tests/vm_nesting.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+
+function parallel() {
+        seq 1 $1 | xargs -P $1 -I "{}" "${@:2}"
+}
+
+function init() {
+	parallel $1 lxc init ubuntu:jammy "t{}" $2
+}
+
+function conf() {
+	parallel $1 lxc config set "t{}" $2
+}
+
+function device() {
+	parallel $1 lxc config device add "t{}" $2 $3 $4
+}
+
+function start() {
+	args=""
+
+	for i in $(seq 1 $1); do
+		args="t$i $args"
+	done
+
+	echo "start $args"
+	lxc start $args
+}
+
+function wait() {
+	parallel $1 bash -c "while true; do if lxc shell t{}; then break; fi; sleep 1; done"
+}
+
+function copy() {
+	parallel $1 lxc file push $2 "t{}$3"
+}
+
+function cmd() {
+	parallel $1 lxc exec "t{}" -- bash -c "$2"
+}
+
+function delete() {
+	args=""
+
+	for i in $(seq 1 $1); do
+                args="t$i $args"
+        done
+
+	echo "delete $args"
+	lxc delete -f $args
+}
+
+if [[ ! -f $1 ]]
+then
+	echo "Specify the path to the LXD binary for sideloading"
+	exit 1
+fi
+
```

Comment:
Can you add a snap install lxd --channel=latest/edge here so that it can be run on a fresh vm